test/position-managers/PositionManager.t.sol

```diff
@@ -0,0 +1,263 @@
+
+        bytes32 positionId =
+            keccak256(abi.encodePacked(address(lpm), range.tickLower, range.tickUpper, bytes32(tokenId)));
+        (uint256 liquidity,,) = manager.getPositionInfo(range.poolKey.toId(), positionId);
+
+        assertEq(liquidity, uint256(params.liquidityDelta) - decreaseLiquidityDelta);
+    }
+
+    function test_decreaseLiquidity_assertCollectedBalance(
+        IPoolManager.ModifyLiquidityParams memory params,
+        uint256 decreaseLiquidityDelta
+    ) public {
+        uint256 tokenId;
+        (tokenId, params) = addFuzzyLiquidity(lpm, address(this), key, params, SQRT_PRICE_1_1, ZERO_BYTES);
+        vm.assume(params.tickLower < 0 && 0 < params.tickUpper); // require two-sided liquidity
+        vm.assume(0 < decreaseLiquidityDelta);
+        vm.assume(decreaseLiquidityDelta < uint256(type(int256).max));
+        vm.assume(int256(decreaseLiquidityDelta) <= params.liquidityDelta);
+
+        LiquidityRange memory range =
+            LiquidityRange({poolKey: key, tickLower: params.tickLower, tickUpper: params.tickUpper});
+
+        // swap to create fees
+        uint256 swapAmount = 0.01e18;
+        swap(key, false, int256(swapAmount), ZERO_BYTES);
+
+        uint256 balance0Before = currency0.balanceOfSelf();
+        uint256 balance1Before = currency1.balanceOfSelf();
+        BalanceDelta delta = decreaseLiquidity(tokenId, decreaseLiquidityDelta, ZERO_BYTES);
+
+        bytes32 positionId =
+            keccak256(abi.encodePacked(address(lpm), range.tickLower, range.tickUpper, bytes32(tokenId)));
+        (uint256 liquidity,,) = manager.getPositionInfo(range.poolKey.toId(), positionId);
+
+        assertEq(liquidity, uint256(params.liquidityDelta) - decreaseLiquidityDelta);
+
+        // The change in balance equals the delta returned.
+        assertEq(currency0.balanceOfSelf() - balance0Before, uint256(int256(delta.amount0())), "boo");
+        assertEq(currency1.balanceOfSelf() - balance1Before, uint256(int256(delta.amount1())), "guh");
+    }
+
+    function test_initialize(IPoolManager.ModifyLiquidityParams memory params) public {
+        // initialize a new pool and add liquidity
+        key = PoolKey({currency0: currency0, currency1: currency1, fee: 0, tickSpacing: 10, hooks: IHooks(address(0))});
+        lpm.initializePool(key, SQRT_PRICE_1_1, ZERO_BYTES);
+
+        params = createFuzzyLiquidityParams(key, params, SQRT_PRICE_1_1);
+
+        // add liquidity to verify pool initialized
+        LiquidityRange memory range =
+            LiquidityRange({poolKey: key, tickLower: params.tickLower, tickUpper: params.tickUpper});
+        mint(range, 100e18, address(this), ZERO_BYTES);
+
+        assertEq(lpm.ownerOf(1), address(this));
+    }
+
+    function test_mintTransferBurn() public {}
+    function test_mintTransferCollect() public {}
+    function test_mintTransferIncrease() public {}
+    function test_mintTransferDecrease() public {}
+    function test_mint_slippageRevert() public {}
```

Comment:
would also love to see
- increase tests
- tests with a 0 delta that just collect fees
- more fee collection tests in general

Reply:
Yeah so the file split rn is:

PositionManager.t.sol -> where we will put all cross combo tests and multicall tests and kind of all integ tests on PositionManger

and then all other test files have specific things they are checking (like increase/fee collecion) in those respective files